A Promsonnet library for creating Prometheus Rules (#358)
grafana/jsonnet-libs: promsonnet/README.md

#PromSonnet

`PromSonnet` is intended as a very simple library for creating Prometheus
alerts and rules. It is 'patching friendly', as in, it maintains the
rules internally as a map, which allows users to easily patch alerts and

> recording rules after the fact. (With lists, this requires iteration, and
> is complex.

Reviewer: ```suggestion
recording rules after the fact, in contrast, lists require complex iteration logic.
```

#PromSonnet

`PromSonnet` is intended as a very simple library for creating Prometheus
alerts and rules. It is 'patching friendly', as in, it maintains the
rules internally as a map, which allows users to easily patch alerts and
recording rules after the fact. (With lists, this requires iteration, and
is complex.

Take this example:

```
local prom = import 'prom.libsonnet';
local promRuleGroupSet = prom.v1.ruleGroupSet;
local promRuleGroup = prom.v1.ruleGroup;
{
  prometheus_up::
    promRuleGroup.new('prometheus_up')
    + promRuleGroup.rule.new(

>       'PrometheusUp', {

Reviewer 2: I'd recommend `PrometheusDown`, as already changed in example.jsonnet. Needs to be updated in this file where applicable.

        alert: 'PrometheusUp',

>         expr: 'up',

Reviewer 2: This should be something like `up{job="prometheus"}==0`, as already changed in example.jsonnet.
